### PR TITLE
Add Enshrouded

### LINKED ISSUE #2675: [Egg Request]: Enshrouded
### Does this expand an already existing service

No

### Link to a game or software

https://enshrouded.com/

### Links for the download

https://enshrouded.zendesk.com/hc/en-us/articles/16051370691485-Dedicated-Server-Installation-on-Steam

### Links for the install docs

https://enshrouded.zendesk.com/hc/en-us/articles/16055441447709-Dedicated-Server-Configuration

### PR DESCRIPTION
**Pull request by QuintenQVD0**
# Description

Closes #2675 

Please keep in mind, this egg is not yet done!
The startup configuration is not yet set. And the config generation will only work if the egg is merged.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [ ] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    "exported_at": "2024-01-23T10:08:55+01:00",
+    "name": "Enshrouded",
+    "author": "josdekurk@gmail.com",
+    "description": "You are Flameborn, last ember of hope of a dying race. Awaken, survive the terror of a corrupting fog, and reclaim the lost beauty of your kingdom. Venture into a vast world, vanquish punishing bosses, build grand halls and forge your path in this co-op survival action RPG for up to 16 players.",
+    "features": null,
+    "docker_images": {
+        "Wine_latest": "ghcr.io\/parkervcp\/yolks:wine_latest"
+    },
+    "file_denylist": [],
+    "startup": "winetricks sound=disabled; xvfb-run --auto-servernum wine \/home\/container\/enshrouded_server.exe",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json:16]: you dont need xvfb and sound disabled

File: game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            "user_viewable": true,
+            "user_editable": false,
+            "rules": "required|numeric",
+            "field_type": "text"
+        },
+        {
+            "name": "WINETRICKS_RUN",
+            "description": "",
+            "env_variable": "WINETRICKS_RUN",
+            "default_value": "vcrun2022 corefonts win10",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json:115]: win10 is not longer needed with wine 9. win10 is default

Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json:115]: stable is still at 8?

File: game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json (hunk 1 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json:16]: won't hurt been there?

File: game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json (hunk 2 above)
Comment (gOOvER) [reply game_eggs/steamcmd_servers/enshrouded/egg-enshrouded.json:115]: stable is on wine 9

**Comment by sebastiangruben** [comment]
Hey, when I use your Enshrouded egg, I get the following error.

```
Executing load_sound disabled
Setting sound driver to disabled
Executing wine C:\windows\syswow64\regedit.exe C:\windows\Temp\set-sound.reg
Executing wine64 C:\windows\regedit.exe C:\windows\Temp\set-sound.reg
wine: failed to open "/home/container/enshrouded_server.exe": c0000135
```

The folder does not contain the enchrouded_server.exe either, maybe something is missing that it was supposed to download?

**Comment by QuintenQVD0** [comment]
> Hey, when I use your Enshrouded egg, I get the following error.
> 
> ```
> Executing load_sound disabled
> Setting sound driver to disabled
> Executing wine C:\windows\syswow64\regedit.exe C:\windows\Temp\set-sound.reg
> Executing wine64 C:\windows\regedit.exe C:\windows\Temp\set-sound.reg
> wine: failed to open "/home/container/enshrouded_server.exe": c0000135
> ```
> 
> The folder does not contain the enchrouded_server.exe either, maybe something is missing that it was supposed to download?

Come to the discord and post install logs.

**Comment by QuintenQVD0** [comment]
**This egg does not allow login, The devs need when they release the game allow anonymous login to download the server**

**Comment by Vapok** [comment]
This egg does not need the xvfb frame buffer.  There's no gui.  The egg that is being socialized on the Enshrouded official Discord is this one: https://github.com/Vapok/pterodactyl-eggs/tree/enshrouded-egg

**Comment by Vapok** [comment]
I'll just wait for this to get merged and then PR my own changes.  Unless you'd like to stop by the Enshrouded Discord and chat it out. :)

**Comment by QuintenQVD0** [comment]
> I'll just wait for this to get merged and then PR my own changes. Unless you'd like to stop by the Enshrouded Discord and chat it out. :)

I would love to chat with you on discord. So feel free to slide in my dm.

**Comment by ionutshiro** [comment]
> Hey, when I use your Enshrouded egg, I get the following error.
> 
> ```
> Executing load_sound disabled
> Setting sound driver to disabled
> Executing wine C:\windows\syswow64\regedit.exe C:\windows\Temp\set-sound.reg
> Executing wine64 C:\windows\regedit.exe C:\windows\Temp\set-sound.reg
> wine: failed to open "/home/container/enshrouded_server.exe": c0000135
> ```
> 
> The folder does not contain the enchrouded_server.exe either, maybe something is missing that it was supposed to download?

Same as him , just tried it

**Pull request closed without merging**